File: examples/jump/simple_mip/simple_mip_test.py
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())

from mos.interface import Interface

def test_simple_mip():

    # Interface
    interface = Interface()

    # Delte model
    interface.delete_model_with_name('Simple MIP JuMP Model')

    # New model
    model = interface.new_model('./examples/jump/simple_mip/simple_mip_model.jl')

    # Existing model by name
    model = interface.get_model_with_name('Simple MIP JuMP Model')

    # Set inputs
    model.set_input_object('lower_bound', 1,2)

    # Solve 
    model.run()

    assert(model.get_status() == 'success')

if __name__ == '__main__':
    test_simple_mip()

File: src/jump/simple_mip/simple_mip_test.jl
import DotEnv
DotEnv.config()

using MOSInterface

@testset "Simple_MIP" begin
    # Interface
    interface = Interface()

    # Delete model
    delete_model_with_name(interface, "Simple MIP JuMP Model")

    # New model
    model = new_model(interface, "./src/jump/simple_mip/simple_mip_model.jl")

    # Get model
    model = get_model_with_name(interface, "Simple MIP JuMP Model")

    # Set inputs
    set_interface_object(model, "lower_bound", 1.2)

    # Show info
    @test get_name(model) == "Simple MIP JuMP Model"
    @test get_system(model) == "jump"
    @test get_status(model) == "created"

    # Solve
    MOSInterface.run(model)

    # Status
    @test get_status(model) == "success"

end    
